issue#206 : Integration test for mutation factory functions

hemant-qlogic opened the pull request:
I tried different criteria for checking the SetCellWOTimestamp test case.
The test case was failing in some cases and passed at some cases.
So as per comment from Carlos, ignored the timestamp while comparing the actual and
expected values.

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+  auto table = CreateTable(table_name, table_config);
+
+  // Create a vector of cell which will be inserted into bigtable
+  std::string const row_key1 = "DelRowKey1", row_key2 = "DelRowKey2";
```
please move `row_key2` to a separate declaration, as in `std::string row_key2 = ...;`.

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+    bulk.emplace_back(std::move(mutation));
+    table.BulkApply(std::move(bulk));
+  }
+  /*
```
Insert blank line here.

coryan replied:
If this is supposed to be a Doxygen comment it is missing the `/**` at the beginning.

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+    bulk.emplace_back(std::move(mutation));
+    table.BulkApply(std::move(bulk));
+  }
+  /*
```
Insert blank line here.

coryan replied:
If this is supposed to be a Doxygen comment it is missing the `/**` at the beginning.

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+      {});
+
+  /// A helper function to create a list of cells.
+  void CreateCellsWithoutTimestamp(bigtable::Table& table,
```
Maybe `CreateCellsWithServerSideTimestamp` ?

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+    table.BulkApply(std::move(bulk));
+  }
+
+  /*
```
If this is supposed to be a Doxygen comment it is missing the `/**` at the beginning.

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+    table.BulkApply(std::move(bulk));
+  }
+
+  /*
```
If this is supposed to be a Doxygen comment it is missing the `/**` at the beginning.

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+    table.BulkApply(std::move(bulk));
+  }
+
+  /*
```
If this is supposed to be a Doxygen comment it is missing the `/**` at the beginning.

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+    table.BulkApply(std::move(bulk));
+  }
+
+  /*
```
If this is supposed to be a Doxygen comment it is missing the `/**` at the beginning.

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+  std::string column_family2 = "fam2";
+  std::string column_family3 = "fam3";
+
+  // Create Table
```
That is not an accurate comment, right?  Maybe:

```C++
/// Define schema for all tables in test.
```

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+
+/**
+ *  Check if the values inserted by SetCell without timestamp are correctly
+ * inserted into db
```
Please consider:

```C++
/**
 * Verify that values inserted with server-side timestamps are correctly inserted into Cloud Bigtable.
```

You may need to reflow that, it probably exceeds the 80 column limit.

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+
+  // Delete the columns with column identifier as created_cells.at(3) (i.e.
+  // column_id2)
+  DeleteFromColumn(*table, created_cells.at(3), timestamp_begin, timestamp_end);
```
This is a bit obscure, why not just `DeleteFromColumn(*table, "column_id3", timestamp_begin, timestamp_end);` ?

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+ * Check if list of table names are matching with the
+ * expected tablename list
+ */
+TEST_F(MutationIntegrationTest, DeleteFromColumnForTimestampValidityTest) {
```
I think this test checks something that is already checked by the unit test, i.e., please remove or rework to *actually* reach the server with an invalid timestamp range.

coryan commented on bigtable/tests/mutation_integration_test.cc:
```diff
@@ -0,0 +1,508 @@
+
+  // Delete the columns with column identifier as created_cells.at(3) (i.e.
+  // column_id2)
+  DeleteFromColumn(*table, created_cells.at(1));
```
Same comment about these function calls being unnecessarily obscure, just say `DeleteFromColumn(*table, column_family2, "column_id3");`

coryan commented:
Did you mean to close this PR before finishing the merge?

hemant-qlogic commented:
I was trying to rebase my changes with the master, So I deleted my commits from my local branch and pushed the changes, may  be due to that the something happened.
Now I changed the file as per your comments and pushed the commits.
Shall I raise the new PR now.
